lib/puppet/module.rb

```diff
@@ -54,7 +55,26 @@ def self.is_module_namespaced_name?(name)
 
   # @api private
   def self.parse_range(range, strict)
-    SemanticPuppet::VersionRange.parse(range, strict)
+    @parse_range_method ||= SemanticPuppet::VersionRange.method(:parse)
+    if @parse_range_method.arity == 1
+      @semver_gem_version ||= SemanticPuppet::Version.parse(SemanticPuppet::VERSION)
+
+      # Give user a heads-up if the desired strict setting cannot be honored
+      if strict
+        if @semver_gem_version.major < 1
+          Puppet.warn_once('strict_version_ranges', 'version_range_cannot_be_strict',
+            _('VersionRanges will never be strict when using non-vendored SemanticPuppet gem, version %{version}') % { version: @semver_gem_version} )
```

Comment:
It looks like there are a few places where `strict_semver` is not getting passed through, so it defaults to `true` and triggers the warning when `--strict_semver` is not specified. For example, given a Gemfile:

```
source "https://rubygems.org"
gem "puppet", :path => '/Users/josh/work/puppet'
gem "semantic_puppet", "~> 0.1.4"
```
Install gems, including old `semantic_puppet`:

```
$ bundle install --path .bundle
Fetching gem metadata from https://rubygems.org/........
Fetching version metadata from https://rubygems.org/.
Resolving dependencies...
Installing CFPropertyList 2.2.8
Installing fast_gettext 1.1.0
Installing locale 2.1.2
Installing text 1.3.1
Installing hiera 3.4.0
Installing hocon 1.2.5
Installing net-ssh 4.1.0
Using bundler 1.14.6
Installing facter 2.4.6 (universal-darwin)
Installing gettext 3.2.3
Installing gettext-setup 0.25
Using puppet 5.0.0 from source at `/Users/josh/work/puppet`
Installing semantic_puppet 0.1.4
Bundle complete! 2 Gemfile dependencies, 13 gems now installed.
Bundled gems are installed into ./.bundle.
```

This is not expected:

```
$ rm -rf ~/.puppetlabs/etc/code/modules
$ bundle exec puppet module install puppetlabs-apache
Notice: Preparing to install into /Users/josh/.puppetlabs/etc/code/modules ...
Notice: Created target directory /Users/josh/.puppetlabs/etc/code/modules
Notice: Downloading from https://forgeapi.puppet.com ...
Warning: VersionRanges will never be strict when using non-vendored SemanticPuppet gem, version 0.1.4
   (file & line not available)
Notice: Installing -- do not interrupt ...
/Users/josh/.puppetlabs/etc/code/modules
└─┬ puppetlabs-apache (v1.11.0)
  ├── puppetlabs-concat (v2.2.1)
  └── puppetlabs-stdlib (v4.17.1)
```

I would only expect the warning when using `puppet module install --strict-semver`. It looks like `Puppet::Forge` calls:

```
        ModuleRelease.new(self, release)
```

which defaults strict_semver to `true` triggering the behavior.